### PR TITLE
New method for setting multiple on selects.

### PR DESCRIPTION
**Pull request by JerseyMilker**
Chaining multiple() method:
`$builder->select('people')->multiple();`

Generates:
`<select name="people[]" multiple="multiple"></select>`;

Great for tags, etc.
Example

### CODE REVIEW

**Comment by adamwathan** [comment]
I dunno if I like automatically appending the brackets, it seems easy enough to do `$builder->select('people[]')->multiple()` and I'd be worried it would be a pain if there was ever a situation where someone didn't want to have the brackets added, or wanted to control it manually.

Right now the package already automatically treats missing method calls as attribute setters, so `multiple('multiple')` does what you want to do here, which makes me think it might be a good idea to always pass the attribute name as the value for the attribute if a missing method is ever called with no parameters, you know?

That would give you this same API:

`$builder->select('people[]')->multiple()`

...without ever having to declare the `multiple` method on the select class.

Basically just means doing this in the `Element` class instead of what we have currently:

```php
    public function __call($method, $params)
    {
        $params = count($params) ? $params : array($method);
        $params = array_merge(array($method), $params);
        call_user_func_array(array($this, 'attribute'), $params);
        return $this;
    }
```

I think I like that solution better personally because it can be applied to things like "required" and stuff too.

If you like that approach feel free to update the PR so you get the credit :)

**Comment by JerseyMilker** [comment]
Hmmm.
`$builder->select('people[]')->multiple()`

**Comment by JerseyMilker** [comment]
Reason I don't like this:
`$builder->select('people[]')->multiple()`

Is because I'm thinking in context of BootForms.  If I do this:
`{!! BootForm::select('Names', 'names[]')->options(['Jesse Leite', 'Adam Wathan'])->multiple('multiple') !!}`
Then 'names[]' with square brackets get rendered into more than just the name attribute:
```<div class="form-group">
	<label class="control-label" for="names[]">Names</label>
	<select name="names[]" id="names[]" class="form-control" multiple="multiple">
		<option value="0">Jesse Leite</option>
		<option value="1">Adam Wathan</option>
	</select>
</div>```
See `id` and `for` attributes. I assume this would make it into bootstrap help blocks for validation handling as well.  Allowing the `multiple()` method to handle the `[]`, means that the other attributes aren't affected when using extensions like BootForm.  Your thoughts?

**Comment by JerseyMilker** [comment]
Reason I don't like this:
```
$builder->select('people[]')->multiple()
```

Is because I'm thinking in context of BootForms.  If I do this:
```
{!! BootForm::select('Names', 'names[]')->options(['Jesse Leite', 'Adam Wathan'])->multiple('multiple') !!}
```

Then 'names[]' with square brackets get rendered into more than just the name attribute:
```
<div class="form-group">
	<label class="control-label" for="names[]">Names</label>
	<select name="names[]" id="names[]" class="form-control" multiple="multiple">
		<option value="0">Jesse Leite</option>
		<option value="1">Adam Wathan</option>
	</select>
</div>
```

See `id` and `for` attributes. I assume this would make it into bootstrap help blocks for validation handling as well.  Allowing the `multiple()` method to handle the `[]`, means that the other attributes aren't affected when using extensions like BootForm.  Your thoughts?

**Comment by JerseyMilker** [comment]
...all that said, I agree with making the param optional:
```
->required()
```
Same as:
```
->required('required')
```
:)

**Comment by JerseyMilker** [comment]
I agree with making the magic method param optional, so that an empty `required()` method is same as `required('required')`, etc.

Reason I don't like this:
```
$builder->select('people[]')->multiple()
```

Is because I'm thinking in context of BootForms.  If I do as you suggest:
```
{!! BootForm::select('Names', 'names[]')->options(['Jesse Leite', 'Adam Wathan'])->multiple('multiple') !!}
```

Then 'names[]' with square brackets get rendered into more than just the name attribute:
```
<div class="form-group">
	<label class="control-label" for="names[]">Names</label>
	<select name="names[]" id="names[]" class="form-control" multiple="multiple">
		<option value="0">Jesse Leite</option>
		<option value="1">Adam Wathan</option>
	</select>
</div>
```

See `id` and `for` attributes. I assume this would affect bootstrap help blocks for validation handling as well.  Allowing the `multiple()` method to handle the `[]`, means that the other attributes aren't affected when using extensions like BootForm.  And more food for thought: When would you ever use `[]` without `multiple="multiple"`, or vice versa?

That said, if you want the option to manually `$builder->select('people[]')`, the Select class' `multiple()` method could regex for `[]` and only add to name if it doesn't already exist in name.  This could be a best of both worlds solution?  Your thoughts?

**Comment by JerseyMilker** [comment]
I committed my suggestion anyway.

This satisfies your requests:
- Add `[]` manually.
- Select's `multiple()` will add `[]` to name, but only if it doesn't exist.
- Attributes added via magic method have optional param as you suggested in your code example.

This satisfies my requests:
- `BootForm::select('people')->multiple()` knows name should have `[]`, but doesn't add `[]` to other attributes, such as id, for, and attributes in Bootstrap help block.

Your thoughts? Let me know when you Form them ;)

File: src/AdamWathan/Form/Elements/Select.php [hunk 1]
```diff
@@ -5,10 +5,13 @@ class Select extends FormControl
 
     private $options;
     private $selected;
+    private $name;
 
     public function __construct($name, $options = array())
     {
+        $this->name = $name;
```
Comment (adamwathan) [inline src/AdamWathan/Form/Elements/Select.php]: Instead of storing this again, just access it with `$this->attributes['name']`. Might be nice in the future to access attributes through `__get` but could be a separate PR.

File: src/AdamWathan/Form/Elements/Select.php [hunk 2]
```diff
@@ -96,4 +99,16 @@ public function defaultValue($value)
         $this->select($value);
         return $this;
     }
+
+    public function multiple()
+    {
+        $name = $this->name;
```
Comment (adamwathan) [inline src/AdamWathan/Form/Elements/Select.php]: Use `$name = $this->attributes['name']`

File: src/AdamWathan/Form/Elements/Select.php [hunk 3]
```diff
@@ -96,4 +99,16 @@ public function defaultValue($value)
...
+        if (substr($name, -2) != '[]') {
+            $name .= '[]';
+        }
```
Comment (adamwathan) [inline src/AdamWathan/Form/Elements/Select.php]: Add a test that covers cases where the name already has `[]` at the end :+1:

**Comment by adamwathan** [comment]
Looks good after addressing my comments above :+1:

File: src/AdamWathan/Form/Elements/Select.php (hunk 3 above)
Comment (JerseyMilker) [inline src/AdamWathan/Form/Elements/Select.php]: Already covered this in `testSelectCanBeMultiple()` test, with two assertions.

**Comment by JerseyMilker** [comment]
Good call, committed change regarding existing name attribute.

Also, I already have an assertion which tests for existing `[]` in multi-select name.  See `SelectTest' > `testSelectCanBeMultiple()`.

**Comment by JerseyMilker** [comment]
<3
